### ISSUE TITLE
Can not find albumName ( Index into NULL Array / UTF8 encoding )

### ISSUE THREAD

**Issue by drmcachr** [TwiggyG2K23/psIMMICH#1]
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/46e82c65-5683-418e-bd0c-b9a48fc67918)

please add more message, I don't know whether API wrong or root directory wrong

**Comment by TwiggyG2K23** [comment]
The problem isn't your folder, there i cant add more messages. It is the try to index in a null array ( property not found ). Presumably you have 0 album. Thats i never tried, at least one always was there. I will fix it :P Until then you can try comment out Set-Strictmode in _params.ps1

**Comment by TwiggyG2K23** [comment]
You can try it now. Use Main branch, pushed in the wrong :P

Added a new function for deleting all albums, needed it for testing now ;) and simple error handling with formatted error outpiut

**Comment by drmcachr** [comment]
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/e38d2811-60ae-4462-be8f-9a37cd6a0640)
and still has utf8 decode issue

**Comment by TwiggyG2K23** [comment]
I think, UTF8 is default with JSON . I will rewrite curl to powershell invoke-webrequest with explicit utf8 encoding. I knew i had to do it at some point :P Hope its working with asian chars.

**Comment by TwiggyG2K23** [comment]
It's your turn :) Switched everything to invoke-webreuqest, whats definitely the better solution, with explicit encoding. Furthermore cleaned the console output and raised the number of the blocksize to 5k at once because of invoke-webrequest hasnt this limitations.

please try this branch. i hope its working. im not the best in encoding questions :(

**Comment by TwiggyG2K23** [comment]
Ok i searched a bit and found https://stackoverflow.com/questions/47952689/powershell-invoke-webrequest-and-character-encoding

Im converted explicid to utf 8 for all put, but no get :P . Maybe the response is a problem. The example asset query above should show what the response. I think that i can try ;) 

Youn can try powershell 7.4 They said, its all utf8 by default.

**Comment by drmcachr** [comment]
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/ceaa657f-d9af-482a-ad5f-0954fe3faa43)

and here is my albums
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/4b26a501-3509-409a-8753-efaa298937a1)

**Comment by drmcachr** [comment]
I try to create album manually, and add a english album, none of them displayed in this command, and the album created by script also contained numbers, these numbers also can not displayed in this command
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/4d5bdb69-002b-40cc-a2e0-f06e389e7e58)

**Comment by TwiggyG2K23** [comment]
Thats curious, for me :(

At first, the response of the one liners shouldnt be empty. But i could reproduce it. Use PowerShell 7.4.

https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.4#installing-the-msi-package

![image](https://github.com/TwiggyG2K23/psIMMICH/assets/141401444/d9d3b0fc-8d5f-4fd5-ad6c-3ea401cd969c)

![image](https://github.com/TwiggyG2K23/psIMMICH/assets/141401444/a9b156dd-2b6e-4991-ab59-86d2764e9768)

![image](https://github.com/TwiggyG2K23/psIMMICH/assets/141401444/addeb86d-2635-4849-a58a-276e8509d2a7)

FingerCrossing.

If its working, i will restore the version without the whole encoding stuff und writing a decent readme :P

**Comment by drmcachr** [comment]
I noticed the windows-terminal I used in windows 11 is version 5

I install powershell 7.4, the command is ok to display albumName
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/13a1f473-f154-4d96-95b3-696a762b2768)
whether chinese or english album I created can display well

but, the main branch still create wrong char albums
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/3328c1e1-1c90-4d46-936b-6d000dc542e9)
and this branch create albums with ????
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/2abd2444-f7bd-471c-92f6-3000fb125e6a)

and here is the response of other command

**Comment by TwiggyG2K23** [comment]
U can use this branch. The command above, with the correct response is without explicit utf8 encoding. its default in powershell 7. Now delete all album, you can use the script and give it a new try ;)

**Comment by drmcachr** [comment]
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/6ecde95a-d82d-4807-af4c-b781231cecf7)
I use powershell 7.4, the album name can displayed now, and I tried main branch, the scripts add album with wrong char, also tried this branch, the scripts add album with ????

**Comment by drmcachr** [comment]
this branch
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/d410c320-a4ce-471f-a025-28a25549416b)

main branch
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/0e9be696-d00b-481c-b64f-70f19374de3c)

**Comment by drmcachr** [comment]
I used latest code in branch curl2web and PowerShell 7.4 here is the result, seems no change in UTF8 encoding
![image](https://github.com/TwiggyG2K23/psIMMICH/assets/13805378/3236ad32-57a8-424c-883d-c645e10e9cba)

**Comment by TwiggyG2K23** [comment]
FRUSTRATING!!! .. but i can reproduce it 

![image](https://github.com/TwiggyG2K23/psIMMICH/assets/141401444/ed293e7d-733c-4499-8eb2-f82045026fb7)

Give me some time

**Comment by TwiggyG2K23** [comment]
I found the problem / difference to the one liner. In the script I used the whole webrequest response ( like curl ) and in the one liner only the content object... whats the right way. I changed it and in my environment with the single special picture name, it works now. Hopefully in yours too. Pls try with update code in this branch and PowerShell 7

![image](https://github.com/TwiggyG2K23/psIMMICH/assets/141401444/5e282524-2ca3-44be-9745-b5bb6c435a7d)

![image](https://github.com/TwiggyG2K23/psIMMICH/assets/141401444/94c623ab-89cd-418e-bf92-0ae9d40511c9)
